core/app/models/spree/tax_category.rb

```diff
@@ -4,7 +4,10 @@ class TaxCategory < Spree::Base
     validates :name, presence: true
     validates_uniqueness_of :name, unless: :deleted_at
 
-    has_many :tax_rates, dependent: :destroy, inverse_of: :tax_category
+
+    has_many :tax_rate_tax_categories, class_name: Spree::TaxRateTaxCategory, dependent: :destroy
```

Comment:
It would be good to add `inverse_of` here since I think that doesn't get added automatically to models in Solidus for some reason.